Foundation/FileManager.swift

```diff
@@ -1027,9 +1049,19 @@ open class FileManager : NSObject {
     /* stringWithFileSystemRepresentation:length: returns an NSString created from an array of bytes that are in the filesystem representation.
      */
     open func string(withFileSystemRepresentation str: UnsafePointer<Int8>, length len: Int) -> String {
-        return NSString(bytes: str, length: len, encoding: String.Encoding.utf8.rawValue)!._swiftObject
+        let encoding = String.Encoding.utf8.rawValue
+        return NSString(bytes: str, length: len, encoding: encoding)!._swiftObject
     }
-    
+
+    open func _string(withFileSystemRepresentation str: UnsafePointer<foundation_char_t>, length len: Int) -> String {
+#if os(Windows)
+      let encoding = String.Encoding.utf16LittleEndian.rawValue
+#else
+      let encoding = String.Encoding.utf8.rawValue
+#endif
```

Comment:
Could you pull this out and put it near the character type alias?  We should just have a `NativeFileSystemEncoding` constant.